### PR TITLE
CASMSEC-344 1.7

### PR DESCRIPTION
**Pull request by alanm-hpe**
## Summary and Scope

Use docker buildkit secrets to propagate `ARTIFACTORY_USERNAME` and `ARTIFACTORY_PASSWORD` into docker build contexts.

## Issues and Related PRs

* Resolves [CASMSEC-344](https://jira-pro.its.hpecorp.net:8443/browse/CASMSEC-344)

### Tested on:

  * Local development environment
  * Jenkins

### Test description:

Built the image locally using an API token for alanm-hpe. Verified the build failed when I revoked the token in algol60 and succeed with a new token.

Verified the built succeeded in algol60 and used the new readonly secret.

## Risks and Mitigations
Low, changing secret propagation

### CODE REVIEW

File: uai-images/basic_uai/zypper.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                  jq \
+                  less \
+                  openssh \
+                  rsync \
+                  tar \
+                  vim \
+                  wget \
+                  which
+
+zypper addrepo --no-gpgcheck -f https://artifactory.algol60.net/artifactory/csm-rpms/hpe/stable/sle-15sp2 algol60
```
Comment (mtupitsyn) [inline uai-images/basic_uai/zypper.sh:27]: FYI - we probably will have to remove anonymous access from https://artifactory.algol60.net/artifactory/csm-rpms as well. So it's probably good time to add `${ARTIFACTORY_USERNAME}:${ARTIFACTORY_PASSWORD}` to this URL as well.

File: uai-images/broker_uai/Dockerfile [hunk 2]
```diff
@@ -23,26 +23,15 @@
...
-  zypper --non-interactive rr --all && \
-  zypper --non-interactive ar ${SLES_MIRROR}/Products/SLE-Module-Basesystem/${SLES_VERSION}/${ARCH}/product?auth=basic sles15sp3-Module-Basesystem-product && \
-  zypper --non-interactive ar ${SLES_MIRROR}/Updates/SLE-Module-Basesystem/${SLES_VERSION}/${ARCH}/update?auth=basic sles15sp3-Module-Basesystem-update && \
-  zypper --non-interactive ar ${SLES_MIRROR}/Products/SLE-Module-Development-Tools/${SLES_VERSION}/${ARCH}/product?auth=basic sles15sp3-Module-Development-Tools-product && \
-  zypper --non-interactive ar ${SLES_MIRROR}/Updates/SLE-Module-Development-Tools/${SLES_VERSION}/${ARCH}/update?auth=basic sles15sp3-Module-Development-Tools-update && \
-  zypper update -y
+COPY zypper.sh /
+RUN --mount=type=secret,id=sles ./zypper.sh && rm /zypper.sh
 
 RUN zypper addrepo --no-gpgcheck -f https://artifactory.algol60.net/artifactory/csm-rpms/hpe/stable/sle-15sp2 algol60
```
Comment (mtupitsyn) [inline uai-images/broker_uai/Dockerfile:29]: Same here, this repo may start require authentication soon...

File: uai-images/basic_uai/zypper.sh (hunk 1 above)
Comment (alanm-hpe) [reply uai-images/basic_uai/zypper.sh]: Done

File: uai-images/broker_uai/Dockerfile (hunk 2 above)
Comment (alanm-hpe) [reply uai-images/broker_uai/Dockerfile]: Done

**Review by erl-hpe — APPROVED** [review]

**Pull request merged**